src/cJSON.c

```diff
@@ -126,16 +126,16 @@ static char *print_number(cJSON *item)
 	if (fabs(((double)item->valueint)-d)<=DBL_EPSILON && d<=INT_MAX && d>=INT_MIN)
 	{
 		str=(char*)cJSON_malloc(21);	/* 2^64+1 can be represented in 21 chars. */
-		if (str) sprintf(str,"%d",item->valueint);
+		if (str) snprintf(str, sizeof(str), "%d",item->valueint);
 	}
 	else
 	{
 		str=(char*)cJSON_malloc(64);	/* This is a nice tradeoff. */
 		if (str)
 		{
-			if (fabs(floor(d)-d)<=DBL_EPSILON && fabs(d)<1.0e60)sprintf(str,"%.0f",d);
-			else if (fabs(d)<1.0e-6 || fabs(d)>1.0e9)			sprintf(str,"%e",d);
-			else												sprintf(str,"%f",d);
+			if (fabs(floor(d)-d)<=DBL_EPSILON && fabs(d)<1.0e60) snprintf(str, sizeof(str), "%.0f",d);
+			else if (fabs(d)<1.0e-6 || fabs(d)>1.0e9)			snprintf(str, sizeof(str), "%e",d);
+			else												snprintf(str, sizeof(str), "%f",d);

```

Comment:
str is a ptr, sizeof(str) is 8bytes on 64bit machine, it is not corrrect

Reply:
I have updated the wrong code with digital number 21 and 64 respectively.